src/Database/FunctionsBuilder.php

```diff
@@ -329,6 +329,24 @@ public function lead(
         return (new AggregateExpression('LEAD', $params, $types, $type ?? 'float'))->over();
     }
 
+    /**
+     * Returns a FunctionExpression representing the Json Value
+     *
+     * @param \Cake\Database\ExpressionInterface|string $expression The Json value or json field
+     * @param string $jsonPath A valid JSON PATH Query
+     * @param array $types list of types to bind to the arguments
+     * @return \Cake\Database\Expression\FunctionExpression
+     */
+    public function jsonValue(
+        ExpressionInterface|string $expression,
+        string $jsonPath,
+        array $types = []
+    ): FunctionExpression {
+        $params = $this->toLiteralParam($expression) + [$jsonPath];
```

Comment:
You want to support passing a sql literal and not an identifier as the field?

Reply:
yes !

Reply:
Can you do this by passing in an expression set to a string?

Reply:
@othercorey How can I do this is this code?

Reply:
In situations where an API should be able to take string as well as identifiers, I'd personally prefer strings to be bound as strings, and require identifiers to be passed as (identifier) expressions.

Reply:
Agree with @ndm2. We can't have any methods that accept both columns names as strings, and string values for a parameter. Doing so makes it impossible for us to prevent SQL injection. If we need to accept string literal values, then column names need to be wrapped in `IdentifierExpression`. I would hope that most usage of JSON value extraction are on columns, as passing a string literal into the database so you can do a json operation on it seems wasteful.

Reply:
@markstory @ndm2 You guys are right ! Thanks !